File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    float lettercount = count_letters(text);

    float wordcount = count_words(text);

    float sentencecount = count_sentences(text);

    int index = round(0.0588 * ((lettercount / wordcount) * 100) - 0.296 * ((sentencecount / wordcount) * 100) - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int lettercount = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            lettercount++;
        }
    }
    return lettercount;
}

int count_words(string text)
{
    int wordcount = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 32)
        {
            wordcount++;
        }
    }
    return wordcount;
}

int count_sentences(string text)
{
    int sentencecount = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentencecount++;
        }
    }
    return sentencecount;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():

    text = get_string("Text: ")

    lettercount = count_letters(text)

    wordcount = count_words(text)

    sentencecount = count_sentences(text)

    index = round(0.0588 * float((lettercount / wordcount) * 100) - 0.296 * float((sentencecount / wordcount) * 100) - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def count_letters(text):
    lettercount = 0
    for i in range(len(text)):
        if text[i].isalpha():
            lettercount += 1
    return lettercount


def count_words(text):
    wordcount = 1
    space = " "
    for i in range(len(text)):
        if text[i] == space:
            wordcount += 1
    return wordcount


def count_sentences(text):
    sentencecount = 0
    end1 = "."
    end2 = "!"
    end3 = "?"
    for i in range(len(text)):
        if text[i] == end1 or text[i] == end2 or text[i] == end3:
            sentencecount += 1
    return sentencecount


if __name__ == "__main__":
    main()